File: process_image.py
from PIL import Image, ImageEnhance
import os

def process_image():
    # Configuration
    input_path = 'assets/images/bio-photo.jpg'
    output_path = 'assets/images/bio-photo-edited.jpg'
    brightness_factor = 1.2  # 20% brightness increase
    
    try:
        print(f"Loading image from {input_path}...")
        
        # Open the image
        img = Image.open(input_path)
        
        # Get dimensions
        width, height = img.size
        
        # Calculate dimensions for square crop
        size = min(width, height)
        left = (width - size) // 2
        top = (height - size) // 2
        right = left + size
        bottom = top + size
        
        # Crop to square
        img_square = img.crop((left, top, right, bottom))
        
        # Increase brightness
        enhancer = ImageEnhance.Brightness(img_square)
        img_bright = enhancer.enhance(brightness_factor)
        
        # Save the processed image
        img_bright.save(output_path)
        
        print(f"Image processed successfully and saved to {output_path}")
        print("To use the new image, you can:")
        print(f"1. Review the edited image at {output_path}")
        print(f"2. If satisfied, replace the original by copying the edited image over {input_path}")
        
    except Exception as e:
        print(f"Error processing image: {e}")

if __name__ == "__main__":
    process_image()

File: process-image.js
const fs = require('fs');
const { createCanvas, loadImage } = require('canvas');

// Configuration
const inputPath = 'assets/images/bio-photo.jpg';
const outputPath = 'assets/images/bio-photo-edited.jpg';
const brightnessAdjustment = 0.2; // 20% brightness increase

async function processImage() {
  try {
    console.log(`Loading image from ${inputPath}...`);
    const image = await loadImage(inputPath);
    
    // Calculate dimensions for square crop
    const sourceSize = Math.min(image.width, image.height);
    const sourceX = (image.width - sourceSize) / 2;
    const sourceY = (image.height - sourceSize) / 2;
    
    // Create canvas with square dimensions
    const canvas = createCanvas(sourceSize, sourceSize);
    const ctx = canvas.getContext('2d');
    
    // Draw the image as a square (cropping to center)
    ctx.drawImage(
      image,
      sourceX, sourceY, sourceSize, sourceSize,
      0, 0, sourceSize, sourceSize
    );
    
    // Apply brightness adjustment
    const imageData = ctx.getImageData(0, 0, sourceSize, sourceSize);
    const data = imageData.data;
    
    for (let i = 0; i < data.length; i += 4) {
      // Adjust RGB values
      data[i] = Math.min(255, data[i] + (255 - data[i]) * brightnessAdjustment);     // Red
      data[i + 1] = Math.min(255, data[i + 1] + (255 - data[i + 1]) * brightnessAdjustment); // Green
      data[i + 2] = Math.min(255, data[i + 2] + (255 - data[i + 2]) * brightnessAdjustment); // Blue
    }
    
    ctx.putImageData(imageData, 0, 0);
    
    // Save the processed image
    const buffer = canvas.toBuffer('image/jpeg');
    fs.writeFileSync(outputPath, buffer);
    
    console.log(`Image processed successfully and saved to ${outputPath}`);
    console.log('To use the new image, rename it to bio-photo.jpg');
  } catch (error) {
    console.error('Error processing image:', error);
  }
}

processImage();